src/Components/Web.JS/src/Platform/Circuits/DefaultReconnectionHandler.ts

```diff
@@ -64,12 +64,20 @@ class ReconnectionProcess {
 
   async attemptPeriodicReconnection(options: ReconnectionOptions) {
     for (let i = 0; i < options.maxRetries; i++) {
-      this.reconnectDisplay.update(i + 1);
+      const currentAttempt = i + 1;
+
+      let retryInterval: number;
+      if (typeof(options.retryIntervalMilliseconds) === 'function') {
+        retryInterval = options.retryIntervalMilliseconds(currentAttempt);
```

Comment:
Also, I think it makes writing the functions a lot easier if the attempt number is zero-indexed. We can just consider it the number of `previousAttempts` rather than the `currentAttempt` number.

```suggestion
        retryInterval = options.retryIntervalMilliseconds(i);
```

Reply:
We could do that, but it then becomes inconsistent with the `update(currentAttempt: number)` method on the `ReconnectDisplay` interface, which customers can already implement today if they want a custom reconnect UI. That existing attempt number is 1-indexed. I agree that zero-indexing probably makes more sense, but I wonder if keeping the consistency is more important in this case.

Reply:
I think renaming the parameter from `currentAttempt` to `previousAttempts` mitigates the apparent inconsistency. You could justify not using `currentAttempt` because it hasn't even happened yet and indeed might not happen if the function being called returns `null`. You could further justify it by saying that `currentAttempt` is more useful to code that's going to use it to render UI, while `previousAttempts` is more useful for code that's going to use it for indexing.

It's not a huge deal though. 1-indexing isn't that bad. Most implementations will just end up subtracting 1. 0-indexing is definitely better.

Reply:
Sure, I'll switch to 0-indexing and rename the parameter to `previousAttempts`